src/components/QuicksightDashboard.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import {
+  QuickSightEmbeddedURLData,
+  LibrariesData,
+  LibraryData,
+} from "../interfaces";
+import { Store } from "redux";
+import { RootState } from "../store";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+
+export interface QuicksightDashboardStateProps {
+  isFetchingLibraries?: boolean;
+  libraries?: LibraryData[];
+}
+
+export interface QuicksightDashboardDispatchProps {
+  fetchLibraries?: () => Promise<LibrariesData>;
+  fetchQuicksightEmbedUrl?: (
+    dashboardId: string,
+    ld: LibrariesData
+  ) => Promise<QuickSightEmbeddedURLData>;
+}
+
+export interface QuicksightDashboardOwnProps {
+  store?: Store<RootState>;
+  dashboardId?: string;
+}
+
+export interface QuicksightDashboardProps
+  extends QuicksightDashboardStateProps,
+    QuicksightDashboardDispatchProps,
+    QuicksightDashboardOwnProps {}
+
+export interface QuicksightDashboardState {
+  embedUrl: string;
+}
+
+class QuicksightDashboard extends React.Component<
+  QuicksightDashboardProps,
+  QuicksightDashboardState
+> {
+  constructor(props) {
+    super(props);
+    this.state = { embedUrl: null };
+  }
+  componentDidMount() {
+    if (this.state.embedUrl) {
+      return;
+    }
+    // Every time the component is mounted a fresh embed url must be fetched.
+    // TODO:  For whatever reason, the "this.props.libraries" variable was not being
+    // set when I tried to put this logic in the render method (nor did it work trying to access it in this method).
+    // Clearly I am missing something. Also the embed url should be set via the action/reducer pattern,
+    // but again I wasn't able to get things working following the pattern in Header.tsx so this is where I landed.
+    // It's ugly but it works.
+    // Nevertheless it should be brought into alignment before it goes into production.
+    this.props.fetchLibraries().then((libs) => {
+      try {
+        this.props
+          .fetchQuicksightEmbedUrl(this.props.dashboardId, libs)
```

Comment:
Assuming that we stop passing the hardcoded value from the parent page, we should replace `this.props.dashboardId` with a new constant in this module.

Reply:
We may offer other dashboards down the road depending on the user's permissions.  However for now, I suppose I can make this a constant.